test/ut/prx_config_ut.c

```diff
@@ -0,0 +1,685 @@
+#define ENABLE_MOCKS
+#include UNIT_C
+
+
+//
+// Simple string compare
+//
+static bool string_is_equal_nocase__hook(
+    const char* val,
+    size_t len,
+    const char* to
+)
+{
+    (void)len;
+    return 0 == strcmp(val, to);
+}
+
+//
+// 3. Setup test suite
+//
+BEGIN_DECLARE_TEST_SUITE()
+REGISTER_UMOCK_ALIAS_TYPE(STRING_HANDLE, void*)
+REGISTER_GLOBAL_MOCK_HOOK(string_is_equal_nocase, string_is_equal_nocase__hook);
+END_DECLARE_TEST_SUITE()
+
+//
+// 4. Setup and run tests
+//
+DECLARE_TEST_SETUP()
+// 
+// Test prx_config_create happy path 
+// 
+TEST_FUNCTION(prx_config_create__success)
+{
+    prx_config_t* config_valid;
+    int32_t result;
+
+    memset(UT_MEM, 0, sizeof(UT_MEM));
+
+    // arrange 
+    STRICT_EXPECTED_CALL(h_realloc(sizeof(prx_config_t), NULL, true, IGNORED_PTR_ARG, IGNORED_NUM_ARG, IGNORED_NUM_ARG))
+        .IgnoreArgument(4).IgnoreArgument(5).IgnoreArgument(6)
+        .SetReturn((void*)UT_MEM);
+
+    // act 
+    result = prx_config_create(&config_valid);
+
+    // assert 
+    ASSERT_EXPECTED_CALLS();
+    ASSERT_ARE_EQUAL(int32_t, er_ok, result);
+}
+
+// 
+// Test prx_config_create passing as created argument an invalid prx_config_t** value 
+// 
+TEST_FUNCTION(prx_config_create__arg_created_null)
+{
+    int32_t result;
+
+    // arrange 
+
+    // act 
+    result = prx_config_create(NULL);
+
+    // assert 
+    ASSERT_EXPECTED_CALLS();
+    ASSERT_ARE_EQUAL(int32_t, er_fault, result);
+}
+
+// 
+// Test prx_config_create unhappy path 
+// 
+TEST_FUNCTION(prx_config_create__neg)
+{
+    prx_config_t* config_valid;
+    int32_t result;
+
+    // arrange 
+    STRICT_EXPECTED_CALL(h_realloc(sizeof(prx_config_t), NULL, true, IGNORED_PTR_ARG, IGNORED_NUM_ARG, IGNORED_NUM_ARG))
+        .IgnoreArgument(4).IgnoreArgument(5).IgnoreArgument(6)
+        .SetReturn(NULL);
+
+    // act 
+    result = prx_config_create(&config_valid);
+
+    // assert 
+    ASSERT_EXPECTED_CALLS();
+    ASSERT_ARE_EQUAL(int32_t, er_out_of_memory, result);
+}
+
+// 
+// Test prx_config_free happy path 
+// 
+TEST_FUNCTION(prx_config_free__success)
+{
+    prx_config_t config_valid;
+
+    for (size_t i = 0; i < _countof(config_valid.values); i++)
+        config_valid.values[i] = (STRING_HANDLE)(0x8000 | i);
+
+    // arrange 
+    for (size_t i = 0; i < _countof(config_valid.values); i++)
+    {
+        STRICT_EXPECTED_CALL(STRING_delete((STRING_HANDLE)(0x8000 | i)));
+    }
+    STRICT_EXPECTED_CALL(h_free(&config_valid, IGNORED_PTR_ARG, IGNORED_NUM_ARG, IGNORED_NUM_ARG))
+        .IgnoreArgument(2).IgnoreArgument(3).IgnoreArgument(4);
+
+    // act 
+    prx_config_free(&config_valid);
+
+    // assert 
+    ASSERT_EXPECTED_CALLS();
+}
+
+// 
+// Test prx_config_free passing as config argument an invalid prx_config_t* value 
+// 
+TEST_FUNCTION(prx_config_free__arg_config_null)
+{
+    // arrange 
+
+    // act 
+    prx_config_free(NULL);
+
+    // assert 
+    // ... 
+    ASSERT_EXPECTED_CALLS();
+}
+
+#if 0
+
```

Comment:
Reminder to get to 90%+ unit test coverage .... An item pushed forward in our backlog...